Human Only Traitor Item

TeomanTheGreat opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a 20tc human only traitor item. Ironically makes them a non-human species but whatever. Also, I'm a tard so, please tell me what to change.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Now humans can flex their superiority even more than they usually do.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added syndicate supersoldiers who are slightly weaker than normal supersoldiers
add: Added it as a human only traitor item
imageadd: Added a syndie nanite injector sprite
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TeomanTheGreat committed:
Syndie Super Soldier

TeomanTheGreat commented:
I know there is a problem with it because it doesn't work when I try to debug and run it so can you guys try and tell me what i need to change.

zeskorion commented on code/modules/mob/living/carbon/human/species_types/syndicate_supersoldier.dm:
```diff
@@ -0,0 +1,19 @@
+/datum/species/human/supersoldier_syndie
```
god, just put this in the supersoldier file and make it a subtype, dammit

TeomanTheGreat replied:
i just did

zeskorion commented on code/modules/mob/living/carbon/human/species_types/syndicate_supersoldier.dm:
```diff
@@ -0,0 +1,19 @@
+	name = "Syndicate Super Soldier" //inherited from the real species, for health scanners and things
+	id = "supersoldier_syndie"
+	species_traits = list(NOTRANSSTING) //all of these + whatever we inherit from the real species
+	inherent_traits = list(TRAIT_NOLIMBDISABLE,TRAIT_NOHUNGER,TRAIT_PIERCEIMMUNE,TRAIT_NODISMEMBER,TRAIT_IGNORESLOWDOWN,TRAIT_IGNOREDAMAGESLOWDOWN,TRAIT_STUNIMMUNE,TRAIT_CONFUSEIMMUNE,TRAIT_SLEEPIMMUNE,TRAIT_VIRUSIMMUNE,TRAIT_NODISMEMBER,TRAIT_THERMAL_VISION,TRAIT_STRONG_GRABBER,TRAIT_ALWAYS_CLEAN,TRAIT_FEARLESS)
```
yeaaaaaaah no

TeomanTheGreat replied:
nerfed it a bit more

zeskorion commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -1707,6 +1707,13 @@ datum/uplink_item/stealthy_tools/taeclowndo_shoes
+	name = "Syndicate Supersoldier Toxin"
+	desc = "A reverse engineered Nanotrasen supersoldier injector. Not as powerful as the Nanotrasen version but will still make you hella robust."
+	item = /obj/item/reagent_containers/hypospray/combat/supersoldier_syndie
+	cost = 20
```
haha it being 20 tc totally makes it balanced right guys

TeomanTheGreat replied:
I'ill give it a pop limit

TeomanTheGreat replied:
please tell me if anything else has to be nerfed

TeomanTheGreat committed:
Adds a Pop cap limit and slightly nerfs the buffs more

TeomanTheGreat commented:
@zeskorion is this good now?

TeomanTheGreat commented:
![image](https://user-images.githubusercontent.com/55861563/84102098-8a37ca00-a9dd-11ea-821c-5f32031da269.png)

zeskorion commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
 		if(light_amount > 0) //if there's any light, heal
 			H.heal_overall_damage(1,1, 0, BODYPART_ORGANIC)
 			H.adjustToxLoss(-1)
-			H.adjustOxyLoss(-1)
```
why are you nerfing the original supersoldier

TeomanTheGreat replied:
did I accidentally delete that?

PowerfulBacon replied:
No, you didn't, it just shows as removed because you removed the end of file from the end of the line. This line is still here right below

zeskorion commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
+	mutanteyes = /obj/item/organ/eyes/night_vision
+	changesource_flags = MIRROR_BADMIN
+
+/datum/species/human/supersoldier_syndie/spec_life(mob/living/carbon/human/H)
```
shouldnt be necessary, make this guy a subtype

zeskorion commented on code/modules/mob/living/carbon/human/species_types/syndicate_supersoldier.dm:
```diff
@@ -0,0 +1,19 @@
+/datum/species/human/supersoldier_syndie
```
you can delete this file

TeomanTheGreat replied:
did i fucking forget to delete this?

zeskorion commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
\ No newline at end of file
+			H.adjustOxyLoss(-1)
+
+/datum/species/human/supersoldier_syndie
```
still isnt a subtype

TeomanTheGreat replied:
zesko mind telling me what a subtype is

TeomanTheGreat committed:
Makes the syndie supersoldier a subtype

TeomanTheGreat committed:
Removing the syndie supersoldier file 

Im a tard forgot to do it

TeomanTheGreat commented:
Doesn't work right now lemme fix

Edit: nevermind im dumb it was because of the pop cap on it

TeomanTheGreat committed:
Fixes a fuck up but it still crashes whenever i inject myself

TeomanTheGreat committed:
Merge remote-tracking branch 'philsmith/Human-Exclusive-Traitor-Item-Super-Juice' into Human-Exclusive-Traitor-Item-Super-Juice

TeomanTheGreat commented:
Ok it gives me this message whenever i try the injector will probably fix tommorow 
![image](https://user-images.githubusercontent.com/55861563/84109609-0ee01380-a9f1-11ea-93e1-dd7f86a28ed2.png)

PowerfulBacon commented:
Does the race even have a sprite?

PowerfulBacon reviewed (commented):
TRAIT_NOLIMBDISABLE,TRAIT_NOHUNGER,TRAIT_NODISMEMBER,TRAIT_IGNORESLOWDOWN,TRAIT_IGNOREDAMAGESLOWDOWN,TRAIT_STUNIMMUNE,TRAIT_CONFUSEIMMUNE,TRAIT_SLEEPIMMUNE,TRAIT_VIRUSIMMUNE,TRAIT_THERMAL_VISION,TRAIT_STRONG_GRABBER,TRAIT_ALWAYS_CLEAN,TRAIT_FEARLESS
Balance on this one is... interesting
 - Cannot be disabled
 - Cannot be stunned
 - No hunger
 - Cannot have their limbs cut off
 - Cannot be slowed down
 - Cannot be stunned
 - Cannot be confused
 - Never needs to sleep 
 - No viruses
 - Thermal vision
 - Night vision
 - Strong grabber (Aggro grabs I think)

As well as healing 0.25 damage damage on life if the light is enough (Always heals if they have their PDA light on).

Super soldier is an admin race for a reason, it doesn't even have a texture.

In my opinion I do not approve of the balance for this one

PowerfulBacon commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
+	id = "supersoldier_syndie"
+	species_traits = list(NOTRANSSTING) //all of these + whatever we inherit from the real species
+	inherent_traits = list(TRAIT_NOLIMBDISABLE,TRAIT_NOHUNGER,TRAIT_NODISMEMBER,TRAIT_IGNORESLOWDOWN,TRAIT_IGNOREDAMAGESLOWDOWN,TRAIT_STUNIMMUNE,TRAIT_CONFUSEIMMUNE,TRAIT_SLEEPIMMUNE,TRAIT_VIRUSIMMUNE,TRAIT_THERMAL_VISION,TRAIT_STRONG_GRABBER,TRAIT_ALWAYS_CLEAN,TRAIT_FEARLESS)
+	mutanteyes = /obj/item/organ/eyes/night_vision
```
This is already inherited from /datum/species/human/supersoldier

PowerfulBacon commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
+/datum/species/human/supersoldier/syndie
+	name = "Syndicate Super Soldier" //inherited from the real species, for health scanners and things
+	id = "supersoldier_syndie"
+	species_traits = list(NOTRANSSTING) //all of these + whatever we inherit from the real species
```
Already inherited from /datum/species/human/supersoldier

PowerfulBacon commented on code/modules/mob/living/carbon/human/species_types/supersoldier.dm:
```diff
@@ -16,4 +16,24 @@
+		var/turf/T = H.loc
+		light_amount = min(1,T.get_lumcount()) - 0.5
+		if(light_amount > 0) //if there's any light, heal
+			H.heal_overall_damage(0.25,0.25, 0, BODYPART_ORGANIC)
```
Having a var/heal_modifier in /datum/species/human/supersoldier and then modifying /datum/species/human/supersoldier/syndie to have heal_modifier = 0.25 would be better than copying the code again.

TeomanTheGreat replied:
i think i managed to do that

PowerfulBacon commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -94,6 +94,13 @@
+	desc = "A reverse engineered nanotrasen supersoldier injector."
+	item_state = "nanite_hypo_syndie"
+	icon_state = "nanite_hypo_syndie"
+	list_reagents = list(/datum/reagent/mutationtoxin/supersoldier/syndie = 5)
```
Could I not take the reagents out of the injector, and make multiple pills of this, making it useable by multiple people?

TeomanTheGreat replied:
i changed the amount of toxin to 1

RedFoxIV commented:
Make it gib lizards on injection

TeomanTheGreat committed:
Nerfs some more 

Reduces the amount of supersoldier toxin to 1 in the injector. Nerfs the buffs even more.

TeomanTheGreat committed:
Fixes shit

TeomanTheGreat committed:
More improvements

The super soldiers will stay invisible because i don't know how to make them keep their looks before mutation.

TeomanTheGreat committed:
Small change i forgot to do in the last commit

TeomanTheGreat commented:
![image](https://user-images.githubusercontent.com/55861563/84220656-4b6b4800-aaa1-11ea-9fc0-666932bb1826.png)
this is still a problem

TeomanTheGreat committed:
Finishes removing the glowing eye thing

being invisible and only a pair of eyes should be obvious enough

TeomanTheGreat committed:
Merge remote-tracking branch 'Upstream/master' into Human-Exclusive-Traitor-Item-Super-Juice

TeomanTheGreat commented:
Please someone tell me where the timer goes I think it has something to do with the toxin itself

TeomanTheGreat commented:
@zeskorion is my pr good yet?

TeomanTheGreat committed:
Revert "Fixes something but not the bug"

This reverts commit cb75fdc1382899097e6227fc25e705dc9d1d1c1f.

TeomanTheGreat committed:
Revert "Revert "Fixes something but not the bug""

This reverts commit 6c53405e7585b2d49382cbefed77dfb9d3e1823c.

TeomanTheGreat committed:
Fixes something but not the bug

eLeCtrOssSnake commented:
No dude, this whole idea is a big unbased meme. Pls no mergg

TeomanTheGreat commented:
Pr is kil
